ray-operator/controllers/ray/common/pod.go

```diff
@@ -689,13 +689,6 @@ func setMissingRayStartParams(rayStartParams map[string]string, nodeType rayv1.R
 	// Add --block option. See https://github.com/ray-project/kuberay/pull/675
 	rayStartParams["block"] = "true"
 
-	// Add dashboard listen port for RayService.
-	if _, ok := rayStartParams["dashboard-agent-listen-port"]; !ok {
-		if value, ok := annotations[utils.EnableServeServiceKey]; ok && value == utils.EnableServeServiceTrue {
-			rayStartParams["dashboard-agent-listen-port"] = strconv.Itoa(utils.DefaultDashboardAgentListenPort)
-		}
-	}
-
```

Comment:
I'm a bit confused about one thing here:

- In this code, if `--dashboard-agent-listen-port` is unspecified by the user, then we set it to `52365`, but only if ServeService is enabled.
- From https://docs.ray.io/en/latest/ray-core/configure.html#all-nodes, it says Ray uses a **random port** if `--dashboard-agent-listen-port` is not specified.
- The RayCluster health probe always uses `52365`: https://github.com/ray-project/kuberay/blob/cbd8adbdd911da7f039fc912137a09e7b044478d/ray-operator/controllers/ray/common/pod.go#L260-L261

So how is the health probe currently working when ServeService is not enabled?

Reply:
Discussed offline.
> From https://docs.ray.io/en/latest/ray-core/configure.html#all-nodes, it says Ray uses a random port if --dashboard-agent-listen-port is not specified.

The doc is incorrect. Ray uses 52365 by default. I will open a PR to update the doc.

> The RayCluster health probe always uses 52365:

It will break if the user specifies `--dashboard-agent-listen-port`.  I will open a followup issue in KubeRay.

Reply:
Ray doc PR: https://github.com/ray-project/ray/pull/42053
KubeRay Issue: https://github.com/ray-project/kuberay/issues/1760